### PR TITLE
feat(wallet): implement ramp network support

### PR DESCRIPTION
**Pull request by Pavneet-Sing**
- Remove wyre

Resolves https://github.com/brave/brave-browser/issues/27678

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

[android-ramp-support.webm](https://user-images.githubusercontent.com/11755381/211047407-8bef867a-4c78-4207-b60e-b63b061becae.webm)

## Additional Info
- [Network selection optimisation](https://github.com/brave/brave-browser/issues/27614)
- [Currency selector](https://github.com/brave/brave-browser/issues/27679)

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java [hunk 1]
```diff
@@ -967,12 +970,11 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
                                 setSendToFromValueValidationResult(errorMessage, false, true);
                             });
                 }
-            } else if (mActivityType == ActivityType.BUY
-                    && mSelectedNetwork.chainId.equals(BraveWalletConstants.MAINNET_CHAIN_ID)) {
+            } else if (mActivityType == ActivityType.BUY) {
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:973]: do we support buy on all networks including test nets? maybe we should exclude test chains in that place?

Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:973]: or it just gives an error `Could not get buy URL` in that case and we are ok?

Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:973]: Those are excluded here with [allowBuy](https://github.com/brave/brave-core/blob/0764b44bf41eb58b3ed76a05989fcaa9d728caa9/android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java#L1636) so the buy option is hidden for excluded networks.

BTW, in future, with more providers, we will have to show a message "buy not supported" based on the result of token list.

Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:973]: Demo for excluded networks

[ramp-excluded-networks.webm](https://user-images.githubusercontent.com/11755381/211055720-bbb0cbc5-1e9f-4514-86de-68d408e4d057.webm)

Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:973]: ok, cool!

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by srirambv** [comment]
Verification passed on Lenovo Tab with Android 12 running 1.49.36 x64 Nightly build
- Verified selecting Buy in the quick action shows `Continue to Ramp` 
- Verified clicking on `Continue to Ramp` loads the on-ramp page with the selected token prefilled
- Verified if no amount is entered a default value is pre-populated on the on-ramp page
- Verified selecting a token and clicking on Buy button shows `Continue to Ramp` to proceed to purchase
- Verified able to select address in dropdown to change the deposit address on the on-ramp page
- Verified on the on-ramp page can't change the token for which it is pre-populated
- Encountered [#27876](https://github.com/brave/brave-browser/issues/27876) & [#27877](https://github.com/brave/brave-browser/issues/27877)

https://user-images.githubusercontent.com/17010094/213089249-6e858650-a438-4532-a5e2-63e0611e757c.mp4

**Pull request merged**